src/mod_time.py

```diff
@@ -0,0 +1,46 @@
+import datetime
+import re
+import importlib
+
+master_module = importlib.import_module('master_module')
+
+
+class ModuleTime(master_module.MasterModule):
+    def __init__(self) -> None:
+        self.time = False
+        self.day = False
+        self.date = False
+
+    def check_command(self, command: str) -> bool:
+        regex = r"\b(dimmi|dammi|che)\b.*\b(ora|ore|giorno|data)\b"
+        if (re.search(regex, command)) is not None:
+            if command.find("ora") != -1 or command.find("ore") != -1:
+                self.time = True
+            if command.find("giorno") != -1:
+                print(command)
+                self.day = True
+            if command.find("data") != -1:
+                self.date = True
+                print(self.time)
+            return True
+        return False
+
+    def execute(self, command: str) -> str:
+        print(command)
+        if self.time:
+            now = datetime.datetime.now()
+            current_time = now.strftime("Sono le : %H:%M")
+            if self.day:
+                d = datetime.datetime.now().strftime(" ed oggi è :%A")
+                current_time += d
+            return current_time
+        if self.day:
+            day = datetime.datetime.now().strftime("%A")
+            print(day)
+            return day
+        if self.date:
+            date = datetime.datetime.now().strftime('Oggi è :%A, %B %d, %Y')
+            print(date)
+            return date
+        return "Specificami la data."
```

Comment:
Questa funzione potrebbe essere snellita notevolmente e ci sono alcune imperfezioni:
1) Il giorno e il mese correnti vengono restituiti in inglese, potrebbe essere utile quindi creare due funzioni esterne che fanno un mapping nome_inglese -> nome_italiano

2) In riga 31 hai definito la variabile **now** che puoi sfruttare (come fatto in riga 32) anche nelle righe 34, 38  e 42 invece di richiamare la funzione `datetime.datetime.now()`

3) Diamo dei nomi coerenti alle variabili, se hai usato **current_time** allora usiamo anche **current_day** e **current_date**

4) Puoi omettere i vari return nei singoli if e dichiarare una stringa **result** che accumula le varie stringhe (in base a ciò che viene richiesto nel comando), in questo modo anche l'if annidato in riga 33 può essere omesso. Quindi poi alla fine fare `return result` (non ho ben chiaro a cosa serva alla fine quel `return "Specificami la data."`, credo possa essere omesso anche quello)